studio/components/layouts/ProjectLayout/LayoutHeader/NotificationsPopover/index.tsx

```diff
@@ -0,0 +1,145 @@
+        .map((notification) => notification.id)
+      if (notificationIds.length > 0) {
+        const { error } = await patch(`${API_URL}/notifications`, { ids: notificationIds })
+        if (error) console.error('Failed to update notifications', error)
+        refresh()
+      }
+    }
+  }
+
+  const onConfirmProjectRestart = async () => {
+    if (!projectToRestart || !targetNotification) return
+
+    const { id } = targetNotification
+    const { id: projectId, ref, region } = projectToRestart
+    const { error } = await post(`${API_URL}/projects/${ref}/restart-services`, {
+      restartRequest: {
+        region,
+        source_notification_id: id,
+        services: ['postgresql'],
+      },
+    })
+
+    if (error) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to restart project: ${error.message}`,
+        error,
+      })
+    } else {
+      app.onProjectPostgrestStatusUpdated(projectId, 'OFFLINE')
+      ui.setNotification({ category: 'success', message: `Restarting project` })
+      router.push(`/project/${ref}`)
+    }
+
+    setProjectToRestart(undefined)
+    setTargetNotification(undefined)
+  }
+
+  return (
+    <>
+      <Popover
+        size="content"
+        align="end"
+        side="bottom"
+        sideOffset={8}
+        onOpenChange={onOpenChange}
+        overlay={
+          <div className="w-[540px]">
+            <div className="flex items-center justify-between border-b border-gray-500 bg-gray-400 px-4 py-2">
+              <p className="text-sm">Notifications</p>
+              {/* Area for improvement: Paginate notifications and show in a side panel */}
+              {/* <p className="text-scale-1000 hover:text-scale-1200 cursor-pointer text-sm transition">
+              See all{' '}
+              {notifications.length > MAX_NOTIFICATIONS_TO_SHOW && `(${notifications.length})`}
+            </p> */}
+            </div>
+            <div className="max-h-[380px] overflow-y-auto py-2">
+              {notifications.length === 0 ? (
+                <div className="py-2 px-4">
+                  <p className="text-scale-1000 text-sm">No notifications available</p>
+                </div>
+              ) : (
+                <>
+                  {notifications.map((notification, i: number) => (
+                    <Fragment key={notification.id}>
+                      <NotificationRow
+                        notification={notification}
+                        onSelectRestartProject={(project, notification) => {
+                          setProjectToRestart(project)
+                          setTargetNotification(notification)
+                        }}
+                      />
+                      {i !== notifications.length - 1 && <Popover.Seperator />}
+                    </Fragment>
+                  ))}
+                </>
+              )}
+            </div>
+          </div>
+        }
+      >
+        <div className="relative">
```

Comment:
There's some extra space underneath the focus state here. An easy fix is to just make it `flex`.
<img width="196" alt="Screen Shot 2022-06-02 at 9 19 56 pm" src="https://user-images.githubusercontent.com/10985857/171785779-9c56e610-785c-4254-bde7-34a4813f0305.png">